spec/features/dataset_show_page_spec.rb

```diff
@@ -101,385 +61,229 @@
       end
     end
   end
 
   feature 'Map preview links' do
     scenario 'WMS Resources have a link to map preview if we have inspire metadata' do
-      dataset = DatasetBuilder
-        .new
-        .with_datafiles(GEO_DATAFILES.map { |f| Datafile.new(f) })
-        .with_inspire_metadata(
-          'bbox_north_lat' => '1.0',
-          'bbox_east_long' => '1.0',
-          'bbox_south_lat' => '2.0',
-          'bbox_west_long' => '2.0',
-        )
-        .build
+      dataset = build :dataset, :inspire
       index_and_visit(dataset)
-
       expect(page).to have_content('Preview on map')
     end
 
     scenario 'WMS Resources have no link to map preview if we have no inspire metadata' do
-      dataset = DatasetBuilder
-        .new
-        .with_datafiles(GEO_DATAFILES.map { |f| Datafile.new(f) })
-        .build
+      dataset = build :dataset
       index_and_visit(dataset)
-
       expect(page).not_to have_content('Preview on map')
     end
   end
 
   feature 'Meta data' do
-    before(:each) do
-      dataset = DatasetBuilder.new.build
-      index_and_visit(dataset)
-    end
-
     scenario 'Display the topic if there is one' do
+      dataset = build :dataset, :with_topic
+      index_and_visit(dataset)
       expect(page).to have_content('Topic: Government')
     end
 
     scenario 'Do not display the topic if information missing' do
-      dataset = DatasetBuilder.new.build
-      dataset['topic'] = nil
-
+      dataset = build :dataset
       index_and_visit(dataset)
-
       expect(page).to have_content('Topic: Not added')
     end
 
     scenario 'Last Updated field displays public_updated_at' do
-      dataset = DatasetBuilder.new.build
+      dataset = build :dataset
+      time = Time.parse(dataset.public_updated_at)
       index_and_visit(dataset)
-      expect(page).to have_content("Last updated: #{dataset['public_updated_at']}")
+      expect(page).to have_content("Last updated: #{time.strftime('%d %B %Y')}")
     end
   end
 
   feature 'Datalinks' do
-    scenario 'displays if required fields present' do
-      dataset = DatasetBuilder.new
-        .with_datafiles(DATA_FILES_WITH_START_AND_ENDDATE)
-        .build
+    let(:dataset) { build :dataset }
 
+    scenario 'displays if required fields present' do
       index_and_visit(dataset)
-
       expect(page).to have_css('h2', text: 'Data links')
     end
 
     scenario 'datafiles are not present' do
-      dataset = DatasetBuilder.new
-        .build
-
       index_and_visit(dataset)
-
       expect(page).to have_content("This data hasn’t been released by the publisher.")
     end
-
-    scenario 'display if some information is missing' do
-      dataset = DatasetBuilder.new
-        .with_datafiles(DATAFILES_WITHOUT_START_AND_ENDDATE)
-        .build
-
-      index_and_visit(dataset)
-
-      expect(page).to have_css('h2', text: 'Data links')
-    end
   end
 
   feature 'Side bar content' do
+    let(:dataset1) { build :dataset, title: '1 Data Set' }
```

Comment:
These datasets are common to all of the following scenarios, so it makes sense to extract them here instead of typing them over again. None of these scenarios require the dataset to have a datafile.